Compilation error when building with boost::sml

**Describe the bug**
A compilation error occurs when building `quill` alongside `boost::sml` on older compilers (GCC 9 on Ubuntu 20.04 and GCC 11 on Ubuntu 22.04). The build fails inside `quill/bundled/fmt/format.h` with an ambiguous overload error for `operator||`.

It happens somewhat like this:
1.  The user code includes `using namespace boost::sml;`. This brings SML's `operator!` and `operator||` into the global lookup scope.
2.  Inside `fmt/format.h` (included in the same file), quill performs a check:
    ```cpp
    if (!is_fast_float<Float>() || is_constant_evaluated()) { ... }
    ```
3. `is_fast_float<Float>()` constructs a temporary object.
    * The compiler matches this temporary object against `boost::sml::operator!(T&&)`.
    * This operator returns a Boost SML expression type.
    * The compiler then attempts to evaluate `SML_Expression || bool`.
    * It attempts to instantiate `boost::sml::operator||`, fails the template constraints (`enable_if`), and errors out because it cannot fall back to the built-in boolean operator effectively.

**To Reproduce**
- repro.cpp
```cpp
// repro.cpp
// Reproducing Steps
// 1. Include Boost.SML
// This library defines aggressive global operator overloads for '!' and '||'
// to support its DSL (e.g., transition + event [!guard]).
#include <boost/sml.hpp>

// 2. Include Quill format.h
#include "quill/bundled/fmt/format.h"

// 3. Pollute namespace
using namespace boost::sml;

int main() {
    return 0;
}
```
- CMakeLists.txt
```
cmake_minimum_required(VERSION 3.14)
project(QuillSmlRepro CXX)

set(CMAKE_CXX_STANDARD 17)

include(FetchContent)

FetchContent_Declare(
  sml
  GIT_REPOSITORY https://github.com/boost-ext/sml
  GIT_TAG        v1.1.11 
)
FetchContent_MakeAvailable(sml)

FetchContent_Declare(
  quill
  GIT_REPOSITORY https://github.com/odygrd/quill.git
  GIT_TAG        master
)
FetchContent_MakeAvailable(quill)

add_executable(repro repro.cpp)
target_link_libraries(repro PRIVATE quill sml)
```
- Dockerfile
```
# Use Ubuntu 20.04 which defaults to GCC 9 or 22.04 which defaults to GCC 11
FROM ubuntu:22.04
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y \
    build-essential \
    cmake \
    git \
    && rm -rf /var/lib/apt/lists/*
WORKDIR /app
COPY CMakeLists.txt repro.cpp ./
RUN mkdir build && cd build && \
    cmake .. && \
    make
```
- Run
```bash
$ docker build .
```
- Get the following error
```bash
8.387 In file included from /app/repro.cpp:9:
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h: In instantiation of 'int fmtquill::v12::detail::format_float(Float, int, const fmtquill::v12::format_specs&, bool, fmtquill::v12::detail::buffer<char>&) [with Float = double]':
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3505:25:   required from 'OutputIt fmtquill::v12::detail::write(OutputIt, T, fmtquill::v12::format_specs, fmtquill::v12::locale_ref) [with Char = char; OutputIt = fmtquill::v12::basic_appender<char>; T = float; typename std::enable_if<fmtquill::v12::detail::is_floating_point<T>::value, int>::type <anonymous> = 0]'
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3516:50:   required from 'OutputIt fmtquill::v12::detail::write(OutputIt, T) [with Char = char; OutputIt = fmtquill::v12::basic_appender<char>; T = float; typename std::enable_if<fmtquill::v12::detail::is_fast_float<T>::value, int>::type <anonymous> = 0]'
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3673:16:   required from 'void fmtquill::v12::detail::default_arg_formatter<Char>::operator()(T) [with T = float; typename std::enable_if<std::integral_constant<bool, (std::is_same<T, int>::value || 1)>::value, int>::type <anonymous> = 0; Char = char]'
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/base.h:2576:52:   required from 'constexpr decltype (vis(0)) fmtquill::v12::basic_format_arg<Context>::visit(Visitor&&) const [with Visitor = fmtquill::v12::detail::default_arg_formatter<char>; Context = fmtquill::v12::context; decltype (vis(0)) = void]'
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format-inl.h:1455:29:   required from here
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: error: no match for 'operator||' in 'boost::ext::sml::v1_1_11::operator!<fmtquill::v12::detail::is_fast_float<double, true> >(fmtquill::v12::detail::is_fast_float<double, true>{std::integral_constant<bool, true>()}) || fmtquill::v12::detail::is_constant_evaluated(0)' (operand types are 'boost::ext::sml::v1_1_11::front::not_<boost::ext::sml::v1_1_11::aux::zero_wrapper<fmtquill::v12::detail::is_fast_float<double, true>, void> >' and 'bool')
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: note: candidate: 'operator||(bool, bool)' (built-in)
8.387 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: note:   no known conversion for argument 1 from 'boost::ext::sml::v1_1_11::front::not_<boost::ext::sml::v1_1_11::aux::zero_wrapper<fmtquill::v12::detail::is_fast_float<double, true>, void> >' to 'bool'
8.387 In file included from /app/repro.cpp:6:
8.387 /app/build/_deps/sml-src/include/boost/sml.hpp:2174:16: note: candidate: 'template<class T1, class T2, typename boost::ext::sml::v1_1_11::aux::enable_if<(boost::ext::sml::v1_1_11::concepts::callable<bool, T>::value && boost::ext::sml::v1_1_11::concepts::callable<bool, T2>::value), int>::type <anonymous> > constexpr auto boost::ext::sml::v1_1_11::operator||(const T1&, const T2&)'
8.387 /app/build/_deps/sml-src/include/boost/sml.hpp:2174:16: note:   template argument deduction/substitution failed:
8.387 /app/build/_deps/sml-src/include/boost/sml.hpp:2173:31: error: no type named 'type' in 'struct boost::ext::sml::v1_1_11::aux::enable_if<false, int>'
8.392 In file included from /app/repro.cpp:9:
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h: In instantiation of 'int fmtquill::v12::detail::format_float(Float, int, const fmtquill::v12::format_specs&, bool, fmtquill::v12::detail::buffer<char>&) [with Float = long double]':
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3505:25:   required from 'OutputIt fmtquill::v12::detail::write(OutputIt, T, fmtquill::v12::format_specs, fmtquill::v12::locale_ref) [with Char = char; OutputIt = fmtquill::v12::basic_appender<char>; T = long double; typename std::enable_if<fmtquill::v12::detail::is_floating_point<T>::value, int>::type <anonymous> = 0]'
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3583:21:   required from 'OutputIt fmtquill::v12::detail::write(OutputIt, T) [with Char = char; OutputIt = fmtquill::v12::basic_appender<char>; T = long double; typename std::enable_if<(fmtquill::v12::detail::is_floating_point<T>::value && (! fmtquill::v12::detail::is_fast_float<T>::value)), int>::type <anonymous> = 0]'
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3673:16:   required from 'void fmtquill::v12::detail::default_arg_formatter<Char>::operator()(T) [with T = long double; typename std::enable_if<std::integral_constant<bool, (std::is_same<T, int>::value || 1)>::value, int>::type <anonymous> = 0; Char = char]'
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/base.h:2578:52:   required from 'constexpr decltype (vis(0)) fmtquill::v12::basic_format_arg<Context>::visit(Visitor&&) const [with Visitor = fmtquill::v12::detail::default_arg_formatter<char>; Context = fmtquill::v12::context; decltype (vis(0)) = void]'
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format-inl.h:1455:29:   required from here
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: error: no match for 'operator||' in 'boost::ext::sml::v1_1_11::operator!<fmtquill::v12::detail::is_fast_float<long double, true> >(fmtquill::v12::detail::is_fast_float<long double, true>{std::integral_constant<bool, false>()}) || fmtquill::v12::detail::is_constant_evaluated(0)' (operand types are 'boost::ext::sml::v1_1_11::front::not_<boost::ext::sml::v1_1_11::aux::zero_wrapper<fmtquill::v12::detail::is_fast_float<long double, true>, void> >' and 'bool')
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: note: candidate: 'operator||(bool, bool)' (built-in)
8.392 /app/build/_deps/quill-src/include/quill/bundled/fmt/format.h:3179:31: note:   no known conversion for argument 1 from 'boost::ext::sml::v1_1_11::front::not_<boost::ext::sml::v1_1_11::aux::zero_wrapper<fmtquill::v12::detail::is_fast_float<long double, true>, void> >' to 'bool'
8.392 In file included from /app/repro.cpp:6:
8.392 /app/build/_deps/sml-src/include/boost/sml.hpp:2174:16: note: candidate: 'template<class T1, class T2, typename boost::ext::sml::v1_1_11::aux::enable_if<(boost::ext::sml::v1_1_11::concepts::callable<bool, T>::value && boost::ext::sml::v1_1_11::concepts::callable<bool, T2>::value), int>::type <anonymous> > constexpr auto boost::ext::sml::v1_1_11::operator||(const T1&, const T2&)'
8.392 /app/build/_deps/sml-src/include/boost/sml.hpp:2174:16: note:   template argument deduction/substitution failed:
8.392 /app/build/_deps/sml-src/include/boost/sml.hpp:2173:31: error: no type named 'type' in 'struct boost::ext::sml::v1_1_11::aux::enable_if<false, int>'
8.419 make[2]: *** [CMakeFiles/repro.dir/build.make:76: CMakeFiles/repro.dir/repro.cpp.o] Error 1
8.419 make[1]: *** [CMakeFiles/Makefile2:145: CMakeFiles/repro.dir/all] Error 2
```
**Expected Behaviour**
The code should build.

**Environment Details**
  - Library Version: master branch/10.1.0
  - Operating System: Ubuntu 20.04 and 22.04.
  - Compiler: GCC9/GCC11

**Additional context**
I was unable to reproduce this issue on Ubuntu 24.04 (GCC 13+). I guess newer GCC versions might handle the SFINAE resolution for the conflicting operators more robustly.

--- a/include/quill/bundled/fmt/format.h
+++ b/include/quill/bundled/fmt/format.h
@@ -3176,7 +3176,7 @@ FMTQUILL_CONSTEXPR20 auto format_float(Float value, int precision,
   int exp = 0;
   bool use_dragon = true;
   unsigned dragon_flags = 0;
-  if (!is_fast_float<Float>() || is_constant_evaluated()) {
+  if (!is_fast_float<Float>::value || is_constant_evaluated()) {
     const auto inv_log2_10 = 0.3010299956639812;  // 1 / log2(10)
     using info = dragonbox::float_info<decltype(converted_value)>;
     const auto f = basic_fp<typename info::carrier_uint>(converted_value);

